langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -80,6 +89,16 @@ public isolated function cloneReadOnly(CloneableType  v) returns CloneableType &
 # - if a record type descriptor specifies default values, these will be used
 #   to supply any missing members
 #
+# ```
+# int[] numberArray = check [1, 2, 3, 4].cloneWithType();
```

Comment:
Let's use a variable for `[1, 2, 3, 4]` with a type that is not `int[]`. This is kinda confusing IMO.

Reply:
Same for some other functions too. The value langlib is a bit different compared to other langlibs because it does conversions.